content/build/building/variables.md

```diff
@@ -0,0 +1,519 @@
+```
+
+Environment variables aren't configurable at build-time by default.
+If you want to change the value of an `ENV` at build-time,
+you can combine environment variables and build arguments:
+
+```dockerfile
+# syntax=docker/dockerfile:1
+
+FROM node:20
+ARG NODE_ENV=production
+ENV NODE_ENV=$NODE_ENV
+WORKDIR /app
+COPY package*.json ./
+RUN npm ci && npm cache clean --force
+COPY . .
+CMD ["node", "app.js"]
+```
+
+With this Dockerfile, you can use `--build-arg` to override the default value of `ENV`:
+
+```console
+$ docker build --build-arg NODE_ENV=development .
+```
+
+Note that, because the environment variables you set persist in containers,
+using them can lead to unintended side-effects for the application's runtime.
+
+For more information on how to use environment variables in builds, refer to:
+
+- [`ENV` Dockerfile reference](../../engine/reference/commandline/builder.md#env)
+
+## Scoping
+
+Build arguments declared in the global scope of a Dockerfile
+aren't automatically inherited into the build stages.
+They're only accessible in the global scope.
+
+```dockerfile
+# syntax=docker/dockerfile:1
+
+# The following build argument is declared in the global scope:
+ARG NAME="joe"
+
+FROM alpine
+# The following instruction doesn't have access to the $NAME build argument
+# because the argument was defined in the global scope, not for this stage.
+RUN echo "hello ${NAME}!"
+```
+
+The `echo` command in this example evaluates to `hello !`
+because the value of the `NAME` build argument is out of scope.
+To inherit global build arguments into a stage, you must consume them:
+
+```dockerfile
+# syntax=docker/dockerfile:1
+
+# Declare the build argument in the global scope
+ARG NAME="joe"
+
+FROM alpine
+# Consume the build argument in the build stage
+ARG NAME
+RUN echo $NAME
+```
+
+Once a build argument is declared or consumed in a stage,
+it's automatically inherited by child stages.
+
+```dockerfile
+# syntax=docker/dockerfile:1
+FROM alpine AS base
+# Declare the build argument in the build stage
+ARG NAME="joe"
+
+# Create a new stage based on "base"
+FROM base AS build
+# The NAME build argument is available here
+# since it's declared in a parent stage
+RUN echo "hello $NAME!"
+```
+
+The following diagram further exemplifies how build argument
+and environment variable inheritance works for multi-stage builds.
+
+{{< figure src="../../images/build-variables.svg" class="invertible" >}}
+
+## Pre-defined build arguments
+
+The following build arguments are pre-defined for all builds.
+You don't need to declare them in your Dockerfile to use them.
+
+- `BUILDPLATFORM`
+- `BUILDOS`
+- `BUILDARCH`
+- `BUILDVARIANT`
+- `TARGETPLATFORM`
+- `TARGETOS`
+- `TARGETARCH`
+- `TARGETVARIANT`
+- `HTTP_PROXY`
+- `http_proxy`
+- `HTTPS_PROXY`
+- `https_proxy`
+- `FTP_PROXY`
+- `ftp_proxy`
+- `NO_PROXY`
+- `no_proxy`
+- `ALL_PROXY`
+- `all_proxy`
+
+The `BUILD*` and `TARGET*` arguments describe the build and target platforms for the build.
```

Comment:
Slightly wondering if `*` here is clear enough for most readers (I understand the approach, but would it be clear to everyone less familiar?)

Reply:
rewrote this bit